### PR TITLE
Update: Added autofix function (fixes #4418)

### LINKED ISSUE #4418: Implement autofixing for comma-dangle rule
<bountysource-plugin>
---
Want to back this issue? **[Post a bounty on it!](https://www.bountysource.com/issues/28244725-implement-autofixing-for-comma-dangle-rule?utm_campaign=plugin&utm_content=tracker%2F282608&utm_medium=issues&utm_source=github)** We accept bounties via [Bountysource](https://www.bountysource.com/?utm_campaign=plugin&utm_content=tracker%2F282608&utm_medium=issues&utm_source=github).
</bountysource-plugin>

### PR DESCRIPTION
**Pull request by maury91**

### CODE REVIEW

**Comment by xjamundx** [comment]
Oh this looks neat! Do you need to update an README file as well to indicate it's autofixable?

**Comment by maury91** [comment]
Thanks for the suggestion, I've updated the documentation about comma dangle.

There's still a problem, travis is not passing, maybe is possible that the test is executed with --fix enabled.

File: lib/rules/comma-dangle.js [hunk 1]
```diff
@@ -122,10 +122,14 @@ module.exports = function(context) {
...
 
         if (trailingToken.value === ",") {
-            context.report(
-                lastItem,
-                trailingToken.loc.start,
-                UNEXPECTED_MESSAGE);
+            context.report({
+                message: UNEXPECTED_MESSAGE,
+                node: lastItem,
+                loc: lastItem.loc.start,
```
Comment (xjamundx) [inline lib/rules/comma-dangle.js]: Looks like this should be `trailingToken.loc.start`

**Comment by maury91** [comment]
Good now travis is passing!

**Comment by xjamundx** [comment]
@maury91 did you sign the CLA yet?

**Comment by ilyavolodin** [comment]
I'm not sure that we should merge this fix. It has potential of conflicting with a lot of other rules. See linked issue for more information.

**Comment by nzakas** [comment]
We can't right now, the issue isn't accepted. We need to wait to see if new autofix changes would allow this to work without conflict.

**Comment by xjamundx** [comment]
@maury91 I'm working on a new home for these types of autofixes that can't really make it into the main repo yet. Would you mind sending your code over here:
https://github.com/xjamundx/eslint-plugin-autofix

It's just an idea right now, but I'll try to get some code in there in the next few hours

**Comment by alberto** [comment]
Closing since we can't do this right now.

**Comment by hzoo** [comment]
I think we can reopen the closed autofix issues given we are accepting PRs for all autofixes now (including this one)

**Comment by alberto** [comment]
@hzoo comma-dangle is now autofixed :) #5925

**Pull request closed without merging**